val.py

```diff
@@ -150,7 +151,7 @@ def run(data,
     # Dataloader
     if not training:
         if device.type != 'cpu':
-            model(torch.zeros(1, 3, imgsz, imgsz).to(device).type_as(next(model.parameters())))  # run once
+            model(torch.zeros(1, 1, imgsz, imgsz).to(device).type_as(next(model.parameters())))  # run once
```

Comment:
maybe some kind of flag for grayscale / normal image - then we can merge grayscale / rgb branches

Reply:
Leaving for now. Subject to change in another task